src/c/core/session/session.c

```diff
@@ -749,13 +752,19 @@ bool wait_session_status(
     {
         send_message(session, buffer, length);
 
-        int64_t start_timestamp = uxr_millis();
-        int remaining_time = UXR_CONFIG_MIN_SESSION_CONNECTION_INTERVAL;
+        const int64_t start_timestamp = uxr_millis();
+        int64_t remaining_time = UXR_CONFIG_MIN_SESSION_CONNECTION_INTERVAL;
+        assert(remaining_time > 0);
```

Comment:
We are not using assert in this project's code base, this can lead to portability issues.

Reply:
Ah, I thought this library was C99 complaint. 

Are the users with incompatible C support able to set NDEBUG all the time then?

If not, can you give an example of the  desired error propagation method?

Reply:
It is preferred to check for the underflow and return false as far as `wait_session_status` was not successful